draft-ietf-webtrans-http3.md

```diff
@@ -366,10 +366,45 @@ an implementation to choose what stream or datagram to discard.
 
 # Session Termination
 
-An WebTransport session over HTTP/3 is terminated when either endpoint closes
-the stream associated with the CONNECT request that initiated the session.
-Upon learning about the session being terminated, the endpoint MUST stop
-sending new datagrams and reset all of the streams associated with the session.
+A WebTransport session over HTTP/3 is considered terminated when either
+endpoint closes the stream associated with the CONNECT request that initiated
+the session.  Upon learning about the session being terminated, the endpoint
+MUST stop sending new datagrams and reset all of the streams associated with
+the session.
+
+To terminate a session with a detailed error message, an application MAY send
+an HTTP capsule [HTTP3-DATAGRAM] of type CLOSE_WEBTRANSPORT_SESSION (0x2843).
+The format of the capsule SHALL be as follows:
+
+~~~
+CLOSE_WEBTRANSPORT_SESSION Capsule {
+  Application Error Code (32),
+  Application Error Message (..8192),
+}
+~~~
+
+CLOSE_WEBTRANSPORT_SESSION has the following fields:
+
+Application Error Code:
+
+: A 32-bit error code provided by the application closing the connection.
+
+Application Error Message:
+
+: A UTF-8 encoded error message string provided by the application closing the
+  connection.  The message takes up the remainer of the capsule, and its
+  length MUST NOT exceed 1024 bytes.
+
+A CLOSE_WEBTRANSPORT_SESSION capsule MUST be followed by a FIN on the sender
+side.  If any additional stream data is received on the CONNECT stream after
+CLOSE_WEBTRANSPORT_SESSION, the stream MUST be reset with code
+H3_MESSAGE_ERROR.  The recipient MUST close the stream upon receiving a FIN.
+If the sender of CLOSE_WEBTRANSPORT_SESSION does not receive a FIN after some
+time, it SHOULD send STOP_SENDING on the CONNECT stream.
```

Comment:
So this isn't an error? Or we think it might be, but we're trying to be more gentle? 
Or we expect it to actually happen in real life if there's a high enough RTT?

Reply:
Maybe this bit of text belongs above (around line 369) and be made generic to the case when a peer sends a FIN (without CLOSE_WEBTRANSPORT_SESSION) as well?

Reply:
Well, there isn't really an "error" when it comes to closing things (since you're closing them anyways).  But yes, in general, (0, "") is the close code unless specified otherwise.